ion_hash_test.go

```diff
@@ -0,0 +1,595 @@
+			err = reader.StepIn()
+			assert.NoError(t, err, "Something went wrong executing reader.StepIn()")
+
+			for reader.Next() {
+				identityUpdateList := [][]byte{}
+				identityDigestList := [][]byte{}
+				identityFinalDigest := [][]byte{}
+				md5UpdateList := [][]byte{}
+				md5DigestList := [][]byte{}
+
+				fieldName = reader.FieldName()
+				hasherName := fieldName
+				if fieldName == "identity" {
+					err = reader.StepIn()
+					assert.NoError(t, err, "Something went wrong executing reader.StepIn()")
+
+					for reader.Next() {
+						annotations := reader.Annotations()
+
+						if len(annotations) > 0 {
+							if annotations[0] == "update" {
+								updateBytes := readSexpAndAppendToList(t, reader)
+								identityUpdateList = append(identityUpdateList, updateBytes)
+
+							} else if annotations[0] == "digest" {
+								digestBytes := readSexpAndAppendToList(t, reader)
+								identityDigestList = append(identityDigestList, digestBytes)
+							} else if annotations[0] == "final_digest" {
+								digestBytes := readSexpAndAppendToList(t, reader)
+								identityFinalDigest = append(identityFinalDigest, digestBytes)
+							}
+						}
+					}
+					err = reader.StepOut()
+					assert.NoError(t, err, "Something went wrong executing reader.StepOut()")
+				} else if fieldName == "md5" {
+					err = reader.StepIn()
+					assert.NoError(t, err, "Something went wrong executing reader.StepIn()")
+
+					for reader.Next() {
+						annotations := reader.Annotations()
+
+						if len(annotations) > 0 {
+							if annotations[0] == "update" {
+								updateBytes := readSexpAndAppendToList(t, reader)
+								md5UpdateList = append(md5UpdateList, updateBytes)
+
+							} else if annotations[0] == "digest" {
+								digestBytes := readSexpAndAppendToList(t, reader)
+								md5DigestList = append(md5DigestList, digestBytes)
+							}
+						}
+					}
+					err = reader.StepOut()
+					assert.NoError(t, err, "Something went wrong executing reader.StepOut()")
+				}
+
+				expectedHashLog := hashLog{
+					identityUpdateList:  identityUpdateList,
+					identityDigestList:  identityDigestList,
+					identityFinalDigest: identityFinalDigest,
+					md5UpdateList:       md5UpdateList,
+					md5DigestList:       md5DigestList,
+				}
+
+				if hasherName != "identity" {
+					testName = testName + "." + hasherName
+				}
+
+				dataList = append(dataList, testObject{testName, testCase, expectedHashLog, newTestIonHasherProvider(hasherName)})
+			}
+			err = reader.StepOut()
+			assert.NoError(t, err, "Something went wrong executing reader.StepOut()")
+		}
+		err = reader.StepOut()
+		assert.NoError(t, err, "Something went wrong executing reader.StepOut()")
+	}
+	return dataList
+}
+
+type testObject struct {
+	hasherName      string
+	testCase        []byte
+	expectedHashLog hashLog
+	provider        *testIonHasherProvider
+}
+
+func writeToWriter(t *testing.T, reader ion.Reader, textWriter ion.Writer, binaryWriter ion.Writer) {
```

Comment:
I suggest moving this function to test_utilities.go where we have a similar function `writeFromReaderToWriter()` defined.

Likewise with any other helper functions that might be useful for tests outside of ion_hash_test.go.

Reply:
Addressed

Reply:
Your latest commit created a new file `test_utilities.go` rather than adding to the already existing `testing_utilities.go`

Reply:
Pardon me. My original comment should have said `testing_utilities.go`

Reply:
I realized I didnt merge upstream, updated and removed test_utilities.go